reports/spec1.tex

```diff
@@ -2,4 +2,55 @@
 
 \section{M1: Spec 1}
 
-TODO\footnote{JK: explain how we translated things}
+\begin{figure}
+    \centering
+    \includegraphics[width=\textwidth,angle=-90]{ffg-callgraph.pdf}
+    \caption{The callgraph of the 3SF specification}
+    \label{fig:your_label}
+\end{figure}
+
+The first specification is obtained by following the principle of least surprise; to the extent that it is possible, without regard to model checking efficiency, we translate Python code to its \tlap{} counterpart with the closest syntax, preserving semantics.
+
+Despite being a programming language and specification language respectively, Python and \tlap{} overlap somewhat in what they can express. 
+For instance, both allow us to express sets, and then filter those sets by a given predicate to obtain a different set.
+To take a concrete example, observe the definition of $\mathsf{pset\_filter}$ in Figure~\ref{py_filter}.
+\begin{figure}
+\begin{lstlisting}[language=Python]
+def pset_filter(p:Callable[[T1],bool],s: PSet[T1]) -> PSet[T1]:
+    r: PSet[T1] = pset()
+
+    for e in s:
+        if p(e):
+            r = r.add(e)
+
+    return r
+\end{lstlisting}
+\caption{$\mathsf{pset\_filter}$ definition \label{py_filter}}
```

Comment:
Is there a reason why you're using math mode?
Otherwise, I'd suggest to clean it here and elsewhere:

```suggestion
\caption{\textsf{pset\_filter} definition \label{py_filter}}
```